### PR TITLE
fix bug on action in connected mode

### PR DESCRIPTION
**Pull request by kirill-knize-sonarsource**

### CODE REVIEW

File: core/src/main/java/org/sonarsource/sonarlint/core/container/storage/StorageAnalyzer.java [hunk 1]
```diff
@@ -75,4 +79,9 @@ public AnalysisResults analyze(ComponentContainer parent, ConnectedAnalysisConfi
     analysisContainer.execute();
     return defaultAnalysisResult;
   }
+
+  public ActiveRule getRule(String ruleKey) {
+    ActiveRules activeRules = analysisContainer.getComponentByType(ActiveRules.class);
```
Comment (henryju) [inline core/src/main/java/org/sonarsource/sonarlint/core/container/storage/StorageAnalyzer.java:90]: NPE if no analysis has been made before.

File: core/src/main/java/org/sonarsource/sonarlint/core/container/storage/StorageAnalyzer.java [hunk 2]
```diff
@@ -61,7 +65,7 @@ private void checkStatus(@Nullable String projectKey) {
   public AnalysisResults analyze(ComponentContainer parent, ConnectedAnalysisConfiguration configuration, IssueListener issueListener, ProgressWrapper progress) {
     checkStatus(configuration.projectKey());
 
-    AnalysisContainer analysisContainer = new AnalysisContainer(parent, progress);
+    analysisContainer = new AnalysisContainer(parent, progress);
```
Comment (henryju) [inline core/src/main/java/org/sonarsource/sonarlint/core/container/storage/StorageAnalyzer.java:74]: I don't think we should do that. The `AnalysisContainer` is intended to be created/disposed for each analysis. You should not keep a reference on it.

File: core/src/main/java/org/sonarsource/sonarlint/core/container/storage/StorageContainer.java [hunk 3]
```diff
@@ -204,29 +204,41 @@ public ConnectedRuleDetails getRuleDetails(String ruleKeyStr) {
...
   }
 
   public ConnectedRuleDetails getRuleDetails(String ruleKeyStr, @Nullable String projectKey) {
     ActiveRule readActiveRuleFromStorage = getHandler().readActiveRuleFromStorage(ruleKeyStr, projectKey);
-    return getRuleDetailsWithSeverity(ruleKeyStr, readActiveRuleFromStorage.getSeverity());
+    String severity;
+    if (readActiveRuleFromStorage != null) {
+      severity = readActiveRuleFromStorage.getSeverity();
+    } else {
+      StorageAnalyzer storageAnalyzer = getComponentByType(StorageAnalyzer.class);
```
Comment (henryju) [inline core/src/main/java/org/sonarsource/sonarlint/core/container/storage/StorageContainer.java:237]: I don't understand why you need to get the severity from the active rule. When the rule come from an extra plugin, there will be no way for the user to override the severity, so you can take the default severity from the rule, using `rulesFromPlugin` like you did earlier.

File: core/src/main/java/org/sonarsource/sonarlint/core/container/storage/StorageContainerHandler.java [hunk 4]
```diff
@@ -107,7 +102,7 @@ public ActiveRule readActiveRuleFromStorage(String ruleKeyStr, @Nullable String
         return activeRulesFromStorage.getActiveRulesByKeyMap().get(ruleKeyStr);
       }
     }
-    throw new IllegalArgumentException("Unable to find active rule with key " + ruleKeyStr);
+    return null;
```
Comment (henryju) [inline core/src/main/java/org/sonarsource/sonarlint/core/container/storage/StorageContainerHandler.java:105]: Then add a `@CheckForNull` to the method

**Review by henryju — CHANGES_REQUESTED** [review]

File: core/src/main/java/org/sonarsource/sonarlint/core/container/storage/SonarQubeActiveRulesProvider.java [hunk 5]
```diff
@@ -95,7 +95,11 @@ private boolean isRuleFromExtraPlugin(Language ruleLanguage, ConnectedGlobalConf
       .stream().anyMatch(extraPluginKey -> ruleLanguage.getLanguageKey().equals(extraPluginKey));
   }
 
-  private static org.sonar.api.batch.rule.ActiveRule createNewActiveRule(ActiveRule activeRule, Sonarlint.Rules storageRules) {
+  public ActiveRules getActiveRules() {
```
Comment (henryju) [inline core/src/main/java/org/sonarsource/sonarlint/core/container/storage/SonarQubeActiveRulesProvider.java:98]: To be deleted

**Review by henryju — APPROVED** [review]
Minor cleanup then LGTM

File: core/src/main/java/org/sonarsource/sonarlint/core/container/storage/SonarQubeActiveRulesProvider.java (hunk 5 above)
Comment (kirill-knize-sonarsource) [reply core/src/main/java/org/sonarsource/sonarlint/core/container/storage/SonarQubeActiveRulesProvider.java]: Done

**Pull request merged**